src/payment/models/buy-fiat/buy-fiat.service.ts

```diff
@@ -86,7 +87,29 @@ export class BuyFiatService {
     });
   }
 
+  async getHistory(userId: number, sellId: number): Promise<BuyFiatHistoryDto[]> {
+    return this.buyFiatRepo
+      .find({
+        where: { sell: { id: sellId, user: { id: userId } } },
+        relations: ['sell', 'sell.user', 'cryptoInput'],
+      })
+      .then((buyFiats) => buyFiats.map(this.toHistoryDto));
+  }
+
   // --- HELPER METHODS --- //
+  private toHistoryDto(buyFiat: BuyFiat): BuyFiatHistoryDto {
+    return {
+      inputAmount: buyFiat.inputAmount,
+      inputAsset: buyFiat.inputAsset,
+      outputAmount: buyFiat.outputAmount,
+      outputAsset: buyFiat.outputAsset,
+      txId: buyFiat.cryptoInput.inTxId,
+      date: buyFiat.cryptoInput.created,
```

Comment:
We should use buyFiat.outputDate here, to be consistent with the other history types.

Reply:
I always tried to pick a date, which is in the very beginning of our processes. Not that we are showing a date to the user with some kind of processing time in between.

For example user is sending it on 1.1.22 at 14:45, we take some time to process or banks take longer and the money went out on 1.1.22 at 17:00. Then I thought we definitely should show 14:45 and not 17:00, as the action of the user was earlier

Reply:
Fair enough. IMO it should be similar for all routes, and for other routes, the outputDate was taken. Perhaps it would help, if we would specifically mark it as completion date? And add the "start" date for other history types as well?

Reply:
yes consistency is key, I will change it for now and we improve in a further update.